ground.py

```diff
@@ -0,0 +1,1309 @@
+        return d
+
+    def getLineageEdge(self, sourceKey):
+        raise NotImplementedError("Invalid call to GroundClient.getLineageEdge")
+
+    def getLineageEdgeLatestVersions(self, sourceKey):
+        raise NotImplementedError(
+            "Invalid call to GroundClient.getLineageEdgeLatestVersions")
+
+    def getLineageEdgeHistory(self, sourceKey):
+        raise NotImplementedError(
+            "Invalid call to GroundClient.getLineageEdgeHistory")
+
+    def getLineageEdgeVersion(self, lineageEdgeId):
+        raise NotImplementedError(
+            "Invalid call to GroundClient.getLineageEdgeVersion")
+
+    ### LINEAGE GRAPHS ###
+    def createLineageGraph(self, sourceKey, name="null", tags=None):
+        d = {
+            "sourceKey": sourceKey,
+            "name": name
+        }
+        if tags is not None:
+            d["tags"] = tags
+        return d
+
+    def createLineageGraphVersion(self, lineageGraphId, lineageEdgeVersionIds, reference=None,
+                                  referenceParameters=None, tags=None, structureVersionId=None, parentIds=None):
+        d = {
+            "lineageGraphId": lineageGraphId,
+            "lineageEdgeVersionIds": lineageEdgeVersionIds
+        }
+        if reference is not None:
+            d["reference"] = reference
+        if referenceParameters is not None:
+            d["referenceParameters"] = referenceParameters
+        if tags is not None:
+            d["tags"] = tags
+        if structureVersionId is not None:
+            d["structureVersionId"] = structureVersionId
+        if parentIds is not None:
+            d["parentIds"] = parentIds
+        return d
+
+    def getLineageGraph(self, sourceKey):
+        raise NotImplementedError("Invalid call to GroundClient.getLineageGraph")
+
+    def getLineageGraphLatestVersions(self, sourceKey):
+        raise NotImplementedError(
+            "Invalid call to GroundClient.getLineageGraphLatestVersions")
+
+    def getLineageGraphHistory(self, sourceKey):
+        raise NotImplementedError(
+            "Invalid call to GroundClient.getLineageGraphHistory")
+
+    def getLineageGraphVersion(self, lineageGraphId):
+        raise NotImplementedError(
+            "Invalid call to GroundClient.getLineageGraphVersion")
+
+    def commit(self, directory=None):
+        return
+
+    def load(self, directory):
+        """
+        This method is implemented by GitImplementation
+        It is used to load the Ground Graph to memory (from filesystem)
+        """
+        return
+
+
+class GitImplementation(GroundAPI):
+    def __init__(self):
+        self.graph = Graph()
+
+    def __run_proc__(self, bashCommand):
+        process = subprocess.Popen(bashCommand.split(), stdout=subprocess.PIPE)
+        output, error = process.communicate()
+        return str(output, 'UTF-8')
+
+        ### EDGES ###
+
+    def createEdge(self, sourceKey, fromNodeId, toNodeId, name="null", tags=None):
```

Comment:
Maybe I'm missing something, but none of these methods are actually using git commands. We're returning heap objects which won't persist across multiple API calls.